storm-client/src/jvm/org/apache/storm/messaging/netty/StormClientHandler.java

```diff
@@ -47,12 +47,20 @@ public void channelRead(ChannelHandlerContext ctx, Object message) throws Except
             BackPressureStatus status = (BackPressureStatus) message;
             if (status.bpTasks != null) {
                 for (Integer bpTask : status.bpTasks) {
-                    remoteBpStatus[bpTask].set(true);
+                    try {
+                        remoteBpStatus[bpTask].set(true);
+                    } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
Could this happen in normal situation? I guess we don't expect it. I don't think catching exception would hurt, but just to be sure about this.

Reply:
I'm also not sure why this can happen?

Reply:
So there are two issues happening in the jira.  Previously when under heavy load a backpressure messsage would be sent.  Under the old code this might contain a -1 which would cause an ArrayIndexOutOFBoundsException to happen.  When it did some backpressure status updates in the message might be lost (specifically the ones that turn off backpressure) and the topology would stop processing.  I fixed the cause of the -1 indexes, but I thought it would be best to also guard against future problems and if there is one we can process the rest of the status updates.

Reply:
OK. Makes sense. I know message of exception will contain the index which will be included to log message, but explicitly write in log message would help users to find the index while grepping since it can be found within same line.

Reply:
Makes sense, thanks.

Reply:
I'm a little surprised that an uncaught exception here doesn't crash the worker though? I'd expect the worker to crash, and for the worker to come back up with backpressure disabled?

Reply:
There is no way to disable backpressure right now.  It didn't crash because this is running under netty.   Netty will catch the exception and pass it off to an exception handler/the future.  I didn't trace down the exact path and why we were not handling this exception to cause it to come down, but I will file a follow on JIRA to examine it.